src/D2L.CodeStyle.Analyzers/Immutability/ImmutableDefinitionChecker.cs

```diff
@@ -474,6 +549,17 @@ private enum AssignmentQueryKind {
 					break;
 			}
 
+			if( assignment.AssignedType == null
+				|| assignment.AssignedType.TypeKind == TypeKind.Error
+			) {
+				diagnostic = Diagnostic.Create(
+					Diagnostics.NonImmutableTypeHeldByImmutable,
+					assignment.Expression.GetLocation(),
+					"blarg", "blarg", "blarg"
+				);
+				return AssignmentQueryKind.Hopeless;

```

Comment:
Need to add some sort of new "Unexpected" diagnostic. (We currently have UnexpectedTypeKind and UnexpectedMemberKind)